### PR TITLE
feat: add container specific securityContext

### PR DESCRIPTION
**Pull request by Demonsthere**
This PR adds the missing container level securityContext to the deployment. 
This option is required to run the operator inside a restricted namespace

### CODE REVIEW

File: charts/opentelemetry-operator/values.yaml [hunk 1]
```diff
@@ -130,6 +130,14 @@ manager:
   # Enable manager pod automatically rolling
   rolling: false
 
+  ## Container specific securityContext
+  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-container
+  securityContext:
```
Comment (TylerHelmuth) [inline charts/opentelemetry-operator/values.yaml:135]: I don't think we need to provide any default values for with securityContext section. Lets set them to `securityContext: {}`.

**Review by TylerHelmuth — COMMENTED** [review]
Thanks for this addition.  Please bnump the patch number of the chart and then run `make generate-examples CHARTS=opentelemetry-operator`

**Review by Allex1 — APPROVED** [review]

**Review by TylerHelmuth — COMMENTED** [review]
@Demonsthere I merged a different operator chart PR first so there are merge conflict nows.  Once they are handled, instead of bumping to `0.30.0` lets bump to `0.29.2`.

**Comment by Demonsthere** [comment]
Conflicts have been addressed :)

**Review by TylerHelmuth — APPROVED** [review]

**Comment by Demonsthere** [comment]
Hey @TylerHelmuth i see some tests are failing but can’t say I see the reason. It does not look to me like it’s related to my changes, or?

**Comment by TylerHelmuth** [comment]
It is flakiness with the operator tests: https://github.com/open-telemetry/opentelemetry-operator/pull/1733

**Pull request merged**